libs/unity/library/Runtime/Scripts/Media/MediaLine.cs

```diff
@@ -51,40 +51,35 @@ public class MediaLine : ISerializationCallbackReceiver
         /// then changing this value raises a <see xref="WebRTC.PeerConnection.RenegotiationNeeded"/> event on the
         /// peer connection of <see cref="Transceiver"/>.
         /// </summary>
-        public IMediaTrackSource Source
+        public MediaTrackSource Source
         {
-            get { return (_source as IMediaTrackSource); }
+            get { return (_source as MediaTrackSource); }
             set
             {
                 if (value == null)
                 {
-                    (_source as IMediaTrackSourceInternal)?.OnRemoveFromMediaLine(this);
-                    _source = null;
-                    DestroySenderIfNeeded();
+                    if (_source != null)
+                    {
+                        _source.OnRemoveFromMediaLine(this);
+                        _source = null;
+                        DestroySenderIfNeeded();
+                    }
                 }
                 else
                 {
                     if (value.MediaKind != MediaKind)
                     {
                         throw new ArgumentException("Wrong media kind", nameof(Source));
                     }
-                    if (!(value is IMediaTrackSourceInternal))
-                    {
-                        throw new ArgumentException("Missing interface IMediaTrackSourceInternal", nameof(Source));
-                    }
-                    if (value is MonoBehaviour mediaTrackSource)
+                    if (_source != value)
                     {
-                        if (_source != mediaTrackSource)
+                        if (_source != null)
                         {
-                            (_source as IMediaTrackSourceInternal)?.OnRemoveFromMediaLine(this);
-                            _source = mediaTrackSource;
-                            (_source as IMediaTrackSourceInternal).OnAddedToMediaLine(this);
-                            CreateSenderIfNeeded();
+                            _source.OnRemoveFromMediaLine(this);
```

Comment:
nit: `_source?.OnRemoveFromMediaLine(this);` without the explicit `if`?

Reply:
`_source` is now a MonoBehaviour and `?.` will give a warning

Reply:
Right, nevermind.